src/shared-components/toggle/index.js

```diff
@@ -16,34 +19,71 @@ const defaultProps = {
   label: '',
 };
 
-const Toggle = ({ checked, label, onChange }) => (
-  <Container>
-    {label && <Label onClick={onChange}>{label}</Label>}
-    <ToggleButton
-      value={checked || false}
-      onToggle={onChange}
-      inactiveLabel=""
-      activeLabel=""
-      thumbStyle={thumbStyle}
-      trackStyle={trackStyle}
-      thumbAnimateRange={[1, 17]}
-      colors={{
-        active: {
-          base: COLORS.secondary,
-        },
-        inactive: {
-          base: COLORS.purple15,
-        },
-        activeThumb: {
-          base: COLORS.white,
-        },
-        inactiveThumb: {
-          base: COLORS.white,
-        },
-      }}
-    />
-  </Container>
-);
+// placeholders for actual theme logic
+const darkTheme = {
+  __type: 'darkTheme',
+  COLORS: {
+    primaryTint1: '#000000',
+    primaryTint5: '#cccccc',
+    secondary: '#414a4c',
+    activeThumb: COLORS.white,
+    inactiveThumb: COLORS.white,
+  },
+};
+
+const lightTheme = {
+  __type: 'lightTheme',
+  COLORS: {
+    primaryTint1: COLORS.primaryTint1,
+    primaryTint5: COLORS.primaryTint3,
+    secondary: COLORS.secondary,
+    activeThumb: COLORS.white,
+    inactiveThumb: COLORS.white,
+  },
+};
+
+const Toggle = ({ checked, label, onChange }) => {
+  const [theme, setTheme] = React.useState(lightTheme);
```

Comment:
This is, of course, not how we'd actually use the theming. But this component was used as an example because:

1. It has clearly distinct colors (a lot of our components are actually just whitespace, so the theming isn't as apparent). 
1. It relies on a third party component to which we're passing colors as props, so not only do we have to account for theming in our `style.ts` files but we need to handle the props we're passing in the component logic. 
1. It highlights a pattern we need to move away from, e.g. using specific colors like `COLORS.purple15`. We will need to migrate all specific colors over to their themed-generics (e.g. `primaryTint5`) so that we can make use of different themes.

Reply:
Good callout on #3 . I wonder how many places are using color constants that are too descriptive about the color itself.